Tests/StoreKitUnitTests/StoreMessagesHelperTests.swift

```diff
@@ -0,0 +1,168 @@
+    private var storeMessagesProvider: MockStoreMessagesProvider!
+
+    private var helper: StoreMessagesHelper!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        self.systemInfo = MockSystemInfo(finishTransactions: true)
+        self.storeMessagesProvider = MockStoreMessagesProvider()
+    }
+
+    @available(iOS 16.4, *)
+    func testShowMessagesAfterDeferMessagesAndNotShowingMessagesAutomaticallyShowsAllDeferredMessages() async throws {
+        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
+        self.createHelper(showStoreMessagesAutomatically: false)
+
+        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
+        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+
+        try await waitForDeferredMessages(messages: [message1, message2])
+
+        await self.helper.showStoreMessages(types: Set(StoreMessageType.allCases))
+
+        expect(message1.displayCalled) == true
+        expect(message2.displayCalled) == true
+    }
+
+    @available(iOS 16.4, *)
+    func testShowMessagesAfterDeferMessagesAndNotShowingMessagesAutomaticallyShowsSpecifiedMessages() async throws {
+        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
+        self.createHelper(showStoreMessagesAutomatically: false)
+
+        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
+        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+
+        try await waitForDeferredMessages(messages: [message1, message2])
+
+        await self.helper.showStoreMessages(types: [StoreMessageType.billingIssue])
+
+        expect(message1.displayCalled) == true
+        expect(message2.displayCalled) == false
+    }
+
+    @available(iOS 16.4, *)
+    func testShowMessagesAfterDeferMessagesAndShowingMessagesAutomaticallyDoesNotShowMessages() async throws {
+        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
+        self.createHelper(showStoreMessagesAutomatically: true)
+
+        let message1 = MockStoreMessage(reason: Message.Reason.billingIssue)
+        let message2 = MockStoreMessage(reason: Message.Reason.priceIncreaseConsent)
+
+        try await waitForDeferredMessages(messages: [message1, message2])
+
+        await self.helper.showStoreMessages(types: Set(StoreMessageType.allCases))
+
+        expect(message1.displayCalled) == false
+        expect(message2.displayCalled) == false
+    }
+
+    @available(iOS 16.4, *)
+    func testShowMessagesAfterDeferMessagesAndNoMessagesDoesNothing() async throws {
+        try AvailabilityChecks.iOS16_4APIAvailableOrSkipTest()
+        self.createHelper(showStoreMessagesAutomatically: true)
+
+        try await waitForDeferredMessages(messages: [])
+
+        await self.helper.showStoreMessages(types: Set(StoreMessageType.allCases))
+    }
+
+    @available(iOS 16.4, *)
+    private func createHelper(showStoreMessagesAutomatically: Bool) {
```

Comment:
Nit: I recommend moving these helpers to an extension of `StoreMessagesHelperTests` so it's easier to find the tests versus the helper methods.